logisland-components/logisland-processors/logisland-processor-common/src/main/java/com/hurence/logisland/processor/SplitRecord.java

```diff
@@ -0,0 +1,144 @@
+                .dynamic(true)
+                .build();
+    }
+    @Override
+    public void init(final ProcessContext context) throws InitializationException {
+        super.init(context);
+    }
+    /*@Override
+    protected Collection<ValidationResult> customValidate(final ValidationContext context) {
+        final List<ValidationResult> validationResults = new ArrayList<>(super.customValidate(context));
+        try {
+
+        } catch (Exception ex) {
+            validationResults.add(
+                    new ValidationResult.Builder()
+                            .input(ex.getMessage())
+                            .valid(false)
+                            .build());
+        }
+        return validationResults;
+    }*/
+
+    @Override
+    public Collection<Record> process(ProcessContext context, Collection<Record> records) {
+        final boolean keepParent = context.getPropertyValue(KEEP_PARENT_RECORD).asBoolean();
+        final boolean keepParentType = context.getPropertyValue(KEEP_PARENT_RECORD_TYPE).asBoolean();
+        final boolean keepParentTime = context.getPropertyValue(KEEP_PARENT_RECORD_TIME).asBoolean();
+        Collection<Record> oringinRecods = new ArrayList<>();
+        try {
+            init(context);
+        } catch (Throwable t) {
+            getLogger().error("error while initializing", t);
+        }
+        try {
+            for (Record record : records) {
+                for (Map.Entry<PropertyDescriptor, String> entry : context.getProperties().entrySet()) {
+                    if (!entry.getKey().isDynamic()) {
+                        continue;
+                    }
+                    String newRecordType = entry.getKey().getName();
+                    String fieldName = entry.getValue();
+                    List<String> fieldNames = getFieldsNames(fieldName);
+                    Map<String, Field> map = new HashMap<>();
+                    Set<Map.Entry<String, Field>> fields = record.getFieldsEntrySet();
+                    for (String field : fieldNames) {
+                        if (record.hasField(field))
+                            map.put(field, record.getField(field));
+                        else throw new IllegalArgumentException("there is no field " + field);
+                    }
+                    Field idField = new Field(record.getId());
+                    map.put("parent_record_id", idField);
+                    long recordTime = 0;
+                    if(!keepParentTime) {
+                        recordTime = System.nanoTime();
+                    } else {recordTime = record.getTime().getTime(); }
+                    String recordType = null;
+                    if (keepParentType) {
+                        recordType = record.getType();
+                    } else {recordType = newRecordType;}
+                    Record newRecord = new StandardRecord().setFields(map).setType(recordType).setTime(recordTime).setId(newRecordType);
```

Comment:
record_id must not be the record_type ! use the GUID by default in StandardRecord